### PR TITLE
Prevent "exec" results from getting GC'd during RegExp.prototype[@@replace]

### PR DESCRIPTION
**Pull request by goyakin**
[VSO bug](https://microsoft.visualstudio.com/web/wi.aspx?pcguid=cb55739e-4afe-46a3-970f-1b49d8ee7564&id=6857149)

### CODE REVIEW

**Comment by goyakin** [comment]
@abchatra, could you take a look?

/cc @tcare

File: lib/Runtime/Library/RegexHelper.cpp [hunk 1]
```diff
@@ -905,40 +905,41 @@ namespace Js
...
-            size_t previousNumberOfCapturesToKeep = 0;
-            Var* captures = nullptr;
+        CompoundString::Builder<64 * sizeof(void *) / sizeof(char16)> accumulatedResultBuilder(scriptContext);
+        CharCount inputLength = input->GetLength();
+        CharCount nextSourcePosition = 0;
+
+        size_t previousNumberOfCapturesToKeep = 0;
+        Var* captures = nullptr;
 
+        BEGIN_TEMP_ALLOCATOR(tempAlloc, scriptContext, _u("RegexHelper"))
```
Comment (abchatra) [inline lib/Runtime/Library/RegexHelper.cpp]: Is the use of tempAlloc in appendReplacement safe?

**Comment by abchatra** [comment]
Looks good

File: lib/Runtime/Library/RegexHelper.cpp (hunk 1 above)
Comment (goyakin) [inline lib/Runtime/Library/RegexHelper.cpp]: It's used to create an array of String indexes for the `ReplaceFormatString` function and won't be GC'd like `results`.

**Pull request merged**